### PR TITLE
Fix the return types of round()/ceil()/floor()

### PR DESCRIPTION
**Pull request by johnbillion**
On PHP < 8 these functions return boolean false when the `$num` parameter is an array: https://3v4l.org/J9KkP

PHPStan correctly reports an error if the type of `$num` is known and not `int|float`, but it's possible for the type not to be known.

This playground example is as good as I could manage: https://phpstan.org/r/59f77cf2-a658-4f2b-9630-80c778714ab2

### CODE REVIEW

**Comment by johnbillion** [comment]
I'm not sure what the failing tests are telling me. They seem unrelated to this change but I don't have a good enough understanding of PHPStan internals to know for sure. Guidance appreciated.

**Comment by ondrejmirtes** [comment]
I'd love to use this, can you please rebase and make the build green? Thank you.

**Comment by ondrejmirtes** [comment]
There's still a lot of failures.

**Comment by johnbillion** [comment]
It's difficult to determine what's failing due to this change and what's also failing in master currently. I'll wait until master is green then take another look.

**Comment by ondrejmirtes** [comment]
Yeah, you can safely rebase now :)

File: src/Type/Php/RoundFunctionReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				'ceil',
+				'floor',
+			],
+			true,
+		);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		if (PHP_VERSION_ID >= 80000) {
```
Comment (ondrejmirtes) [inline src/Type/Php/RoundFunctionReturnTypeExtension.php:40]: You need to use `PhpVersion` injected to constructor, and call a specific method on it. Something like `hasStricterRoundFunctions()` or something like that.

File: src/Type/Php/RoundFunctionReturnTypeExtension.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			return $noArgsReturnType;
+		}
+
+		$firstArgType = $scope->getType($functionCall->getArgs()[0]->value);
+
+		if ($firstArgType instanceof MixedType) {
+			return $defaultReturnType;
+		}
+
+		if (PHP_VERSION_ID >= 80000) {
```
Comment (ondrejmirtes) [inline src/Type/Php/RoundFunctionReturnTypeExtension.php:68]: dtto

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by johnbillion** [comment]
Thanks for helping get it over the finishing line :-)

**Pull request merged**